### ISSUE TITLE
analysis.errors event has incorrect file parameter for a library that has a missing part file

### ISSUE THREAD

**Issue by skybrian** [dart-lang/sdk#25826]
If a part file is missing, the analysis.errors event sometimes has the wrong filename.

Here is a file to reproduce the error:

```
part "does_not_exist.dart";

_foo() => [];
```

The hint is:

```
The function '_foo' is not used - no_part_file.dart, line 3
```

However, the analysis.errors event has a "file" parameter with a filename ending with "does_not_exist.dart":

Here is the protocol message:

```
{file:  /var/.../tmp/1/does_not_exist.dart,
  errors: [{severity: INFO, type: HINT, location: {
  file: /var/.../tmp/1/no_part_file.dart, offset: 29, length: 4, startLine: 1, startColumn: 30},
  message: The function '_foo' is not used, hasFix: false}]}
```

(Found while debugging a stress test.)

**Comment by skybrian** [comment]
Possibly relevant: in the stress test, this is using an overlay only. (No file is saved to disk.) I also noticed that when the stress test modifies the filename in the part statement, the error from before isn't cleared. So as far as the client is concerned, there are errors attached to a nonexistent file that is no longer referenced anywhere, either.

**Comment by skybrian** [comment]
I just ran into something similar while working on a side project.

While editing a part file in Atom, after adding a new unused field:

  StreamSubscription _onScrollSub;

The hint is repeated six times:
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 12
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 14
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 13
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 12
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 15
The value of the field '_onScrollSub' is not used - lib/tagtree/widgets/tagplace.dart, line 12

There is actually one declaration, on line 12.
It so happens that there are five parts in this library, for six total files. Reanalyzing sources or restarting the analysis server doesn't make this go away.

I tried adding another part to the library, and that results in seven warnings. Removing the part doesn't immediately change anything,  but if I reanalyze sources it drops down to six.

Running dartanalyzer from the command line reports only one hint.
